### ISSUE TITLE
Dbt manifest node source ingestion overwriting node ingestion

### ISSUE THREAD

**Issue by jeremytee97** [open-metadata/OpenMetadata#13286]
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
- Ingestion Framework

**Describe the bug**

What happened:
The dbt tab in the UI (which shows the sql query of a model) is missing when we perform ingestion for different dbt projects which have dependency on each other.

Scenario:
Trigger dbt ingestion for project A, dbt tab of model A in project A is available
Trigger dbt ingestion for project B, dbt tab of model A in project A is gone
Overall lineage: model A in project A >> model B in project B (model A in project A is a source for model B in project B)

The potential root cause(what we think):
OMD will process both nodes and sources info during dbt_manifest.json ingestion at [here](https://github.com/open-metadata/OpenMetadata/blob/1.1.1-release/ingestion/src/metadata/ingestion/source/database/dbt/metadata.py#L316-L398) and causing overwriting.
 In other words, the dbt manifest of project B only contains the source info of model A with empty SQL code and caused the dbt tab of model A to go missing.
Ideal case:
When we perform dbt ingestion for project B, it should only reflect the changes in the lineage diagram of model A in project A and not overwrite its dbt tab info.
Our OMD version: v1.1.1 (edited)

Slack Thread: https://openmetadata.slack.com/archives/C02B6955S4S/p1695200973222229

**Comment by ekng94** [comment]
i believe the impact of this bug :bug: is more than just affecting the dbt tab alone.
Today we encountered the overwriting of the source table description as per scenario below:

**Scenario:**
1. `dbtUpdateDescriptions` is set to `true`
2. Trigger dbt ingestion for `dbt project A`
3. Table description of s`ource tables` for `dbt project A` got updated

**Root cause:**
1. the `source.yaml `in `dbt project A` comes with description

**Ideal case:**
1. Source info should not be overwritten

**Comment by ekng94** [comment]
seems like a better solution is to process both `nodes` and `sources` info with different logic during dbt ingestion

**Comment by jeremytee97** [comment]
Any updates on this @OnkarVO7 ?
